base_layer/core/src/base_node/state_machine_service/states/horizon_state_sync/horizon_state_synchronization.rs

```diff
@@ -411,14 +422,54 @@ impl<'a, B: BlockchainBackend + 'static> HorizonStateSynchronization<'a, B> {
 
         let header = self.shared.db.fetch_chain_header(self.horizon_sync_height).await?;
         let horizon_data = self.db().fetch_horizon_data().await?.unwrap_or_else(HorizonData::zero);
-        let mut pruned_kernel_sum = horizon_data.kernel_sum().clone();
         let mut pruned_utxo_sum = horizon_data.utxo_sum().clone();
-        let chain_metadata = self.db().get_chain_metadata().await?;
+        let mut pruned_kernel_sum = HomomorphicCommitment::default();
 
-        for h in chain_metadata.pruned_height()..=header.height() {
+        let mut prev_mmr = 0;
+        let mut prev_kernel_mmr = 0;
+        for h in 0..=header.height() {
             let accum_data = self.db().fetch_block_accumulated_data_by_height(h).await?;
-            pruned_kernel_sum = accum_data.kernel_sum() + &pruned_kernel_sum;
-            pruned_utxo_sum = accum_data.utxo_sum() + &pruned_utxo_sum;
+            let curr_header = self.db().fetch_chain_header(h).await?;
+
+            debug!(
+                target: LOG_TARGET,
+                "Fetching utxos from db: height:{}, header.output_mmr:{}, prev_mmr:{}, end:{}",
+                curr_header.height(), curr_header.header.output_mmr_size, prev_mmr, curr_header.header.output_mmr_size -1);
+            let utxos = self
+                .db()
+                .fetch_utxos_by_mmr_position(prev_mmr, curr_header.header.output_mmr_size - 1, header.hash().clone())
+                .await?;
+            let mut utxo_sum = HomomorphicCommitment::default();
+            debug!(target: LOG_TARGET,
+            "Number of utxos returned: {}", utxos.0.len());
+            for u in utxos.0 {
+                match u {
+                    PrunedOutput::NotPruned { output } => {
+                        utxo_sum = &output.commitment + &utxo_sum;
+                    },
+                    _ => {
+                        trace!(target: LOG_TARGET, "Pruned output ignored");
+                    },
+                }
+            }
+            prev_mmr = curr_header.header.output_mmr_size;
+
+            pruned_utxo_sum = &utxo_sum + &pruned_utxo_sum;
+
+            debug!(
+                target: LOG_TARGET,
+                "Fetching kernels from db: height:{}, header.kernel_mmr:{}, prev_mmr:{}, end:{}",
+                curr_header.height(), curr_header.header.kernel_mmr_size, prev_kernel_mmr, curr_header.header.kernel_mmr_size -1);
+            let kernels = self.db().fetch_kernels_by_mmr_position(prev_kernel_mmr, curr_header.header.kernel_mmr_size -1 ).await?;
+
+            debug!(target: LOG_TARGET,
+                   "Number of kernels returned: {}", kernels.len());
+            for k in kernels {
+                pruned_kernel_sum = &k.excess + &pruned_kernel_sum;
+
+            }
+            prev_kernel_mmr = curr_header.header.kernel_mmr_size;
```

Comment:
We dont have to do this for the kernels, we already save the kernel sum per block in the `block_accumulated_data` which is already saved in this loop as `accum_data `

We can replace this for loop with:
`pruned_kernel_sum = accum_data.kernel_sum() + &pruned_kernel_sum;`